### PR TITLE
UI menus

### PR DESCRIPTION
**Pull request by ikeia**
- Title Screen
-  Game over screen
- Bullet effects
- Upgrade screen
- Bullet indicators
- Turret model
- Enemies left

### CODE REVIEW

File: Assets/Scripts/Enemy/EnemyBehaviour.cs [hunk 1]
```diff
@@ -64,10 +64,11 @@ IEnumerator Cooldown()
     // Update is called once per frame
     void Update()
     {
-        if (!_initialized) return;
+        if (!_initialized || TimeBody.isRewinding) return;
```
Comment (zhuandy42) [inline Assets/Scripts/Enemy/EnemyBehaviour.cs]: unnecessary for now

File: Assets/Scripts/Player/Movement.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@ public class Movement : MonoBehaviour
 {
     // Movement variables
     public static float speed = 12f;
-    public static float dashSpeed = 3f;
+    public static float dashSpeed = 5f;
```
Comment (zhuandy42) [inline Assets/Scripts/Player/Movement.cs]: conflicting with dash changes

File: Assets/Scripts/Player/PlayerCentral.cs [hunk 3]
```diff
@@ -32,47 +40,60 @@ void Start()
...
-        }
+        //Jump
+        if (Input.GetButtonDown("Jump") && isGrounded) _velocity.y += Movement.jumpVelocity;
 
-        if (Input.GetKey(KeyCode.LeftShift))
-        {
-            Movement.playerDash(_player);
-        }
+        //Dash
+        if (Input.GetKey(KeyCode.LeftShift)) Movement.playerDash(_player);
```
Comment (zhuandy42) [inline Assets/Scripts/Player/PlayerCentral.cs]: conflicting with dash changes

File: Assets/Scripts/Stats.cs [hunk 4]
```diff
@@ -12,7 +12,22 @@ public float currentHealth
         get => _currentHealth;
         set {
             _currentHealth = value;
-            if (value <= 0) Destroy(gameObject);
+            if (value <= 0)
+            {
+                if (gameObject.tag == "Player")
```
Comment (zhuandy42) [inline Assets/Scripts/Stats.cs:17]: should probably move this logic elsewhere, stats is shared among objects

File: Assets/Scripts/Stats.cs [hunk 5]
```diff
@@ -76,7 +92,19 @@ public void takeDamage(float damage)
         setHealth(currentHealth - damage);
         if (currentHealth <= 0)
         {
-            Destroy(gameObject);
+            if (gameObject.tag != "Player")
```
Comment (zhuandy42) [inline Assets/Scripts/Stats.cs:95]: same

File: Assets/Scripts/Weapons/BulletAttackBehaviour.cs [hunk 6]
```diff
@@ -40,6 +40,7 @@ public virtual void onHit(BulletMono bm, GameObject target)
             if (statsComponent.owner != _owner)
             {
                 statsComponent.takeDamage(_damage);
+                statsComponent.currentHealth -= _damage;
```
Comment (zhuandy42) [inline Assets/Scripts/Weapons/BulletAttackBehaviour.cs:43]: if adding this, remove takeDamage

**Review by zhuandy42 — APPROVED** [review]
lgtm

**Review by MaXeraph — APPROVED** [review]
LGTM @ethanakester

**Pull request merged**